docs/user-guide/CAI.rst

```diff
@@ -0,0 +1,308 @@
+
+A harmless but non-evasive assistant: reducing the tension between helpfulness and harmlessness, and avoiding evasive responses that reduce transparency and helpfulness.
+
+Simplicity and transparency: encoding the training goals in a simple list of natural language instructions or principles, and using chain-of-thought reasoning to make AI decision making explicit and understandable.
+
+Reducing iteration time: obviating the need to collect new human feedback labels when altering the objective or testing different behaviors.
+
+Train a CAI model
+#####################
+
+This section is a step-by-step tutorial that walks you through how to run a full CAI pipeline with a ``Mistral-7B`` LLM model. It includes the following:
+
+1. Data download and preprocessing.
+
+2. Generate responses to harmfulness prompts using a helpful-only AI assistant. Ask the model to critique its response according to a principle in the constitution, and then revise the original response in light of the critique.
+
+3. Finetune ``Mistral-7B`` with SFT on the revised responses to create a ``Mistral-7B-SL-CAI`` model.
+
+4. 
+   a. Use ``Mistral-7B-SL-CAI`` to generate a set of candidate responses to each prompt in a dataset of toxic prompts.
+   b. Formulate each prompt and pair into a multiple choice question, where we ask ``Mixtral-8x7B`` which response is best according to the constitution.
+   c. Blend the AI feedback preference dataset (prompts and pairs) with human feedback helpfulness dataset.
+
+5. Train a RM.
+
+6. Finetune the ``Mistral-7B-SL-CAI`` with PPO and the RM to train a ``Mistral-7B-RL-CAI`` model.
+
+7. Inference.
+
+.. image:: ../assets/cai_flow.png
+
+Step 1: Download models and datasets
+#############################################################################
+1. Download ``Mistral-7B-Instruct`` and ``Mistral-7B`` LLM models from https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1 and https://huggingface.co/mistralai/Mistral-7B-v0.1 into the models folder.
+
+   Then convert into .nemo format:
+   
+   .. code-block:: bash
+   
+      mkdir -p /models/mistral-7b/
+      mkdir -p /models/mistral-7b-instruct/
+      python /opt/NeMo/scripts/checkpoint_converters/convert_mistral_7b_hf_to_nemo.py --in-file /path/to/mistral-7b --out-file /models/mistral/mistral-7b.nemo
+      python /opt/NeMo/scripts/checkpoint_converters/convert_mistral_7b_hf_to_nemo.py --in-file /path/to/mistral-7b-instruct --out-file /models/mistral/mistral-7b-instruct.nemo
+
+
+2. Download the red teaming dataset:
+
+   .. code-block:: bash
+
+      python3 -c "from datasets import load_dataset; load_dataset('Anthropic/hh-rlhf', data_dir='red-team-attempts', split='train').to_json('/path/to/anthropic_red_team_attempts_train.json')"
+   
+   This will download the dataset to ``/path/to/anthropic_red_team_attempts_train.json``
```

Comment:
This command will download the dataset to ``/path/to/anthropic_red_team_attempts_train.json``.